### PR TITLE
Reorganize articles into more suitable categories

### LINKED ISSUE #107: Suggestions for grouping articles in the sidebar
I have a few suggestions for nesting/grouping articles in a way that I think makes more sense:

- Move `Databases` and `Security` from `Other Topics` to `SE Topics`

- Move `Static Analysis` into `Code Quality`

- Rename `SE Topics` to `Software Engineering Topics`

- Remove `Project Management` and move `Agile Development` and `DevOps` into SE Topics (DevOps isn't related to project management)

- Remove `Tools Useful for Software Engineers` and move `Regular Expressions` into `SE Topics`

- Move `Writing Testable Code` from `Testing` into `Code Quality` (the contents of that chapter are more closely related to code quality than testing)

- Rename `Architecture` to `Software Architecture`

- Move `Scalability` into `Performance` and rename the category to `Scalability and Performance`

- Swap the order of `SE Topics` and `App Frameworks`

  So that the overall structure flows more smoothly:
  1. Programming Languages
  2. App Frameworks
  3. SE Topics
  4. Other Topics

### PR DESCRIPTION
**Pull request by luyangkenneth**
Closes #107 - this PR implements all the suggestions in that issue

### CODE REVIEW

**Comment by damithc** [comment]
Open for peer reviews ...

**Comment by luyangkenneth** [comment]
I will also announce this later on, so that everyone can fix their merge conflicts (if any) based off this issue/PR.

**Comment by damithc** [comment]
Should we hold this back until almost all the current articles are in? This structure might even need further tweaks, for example, there are some database topics current WIP.

**Comment by luyangkenneth** [comment]
I think it'll be less painful to incorporate these changes early. I'm also happy to do another smaller round of tweaks if necessary.

That said, I'm certainly mindful about breaking others' workflows, so I've scanned through all the open issues/PRs, and the number of conflicts I foresee seem to be pretty minimal 😃

I can't seem to find any database-related issues/PRs at the moment - could you point them out to me please?

**Comment by damithc** [comment]
> I can't seem to find any database-related issues/PRs at the moment - could you point them out to me please?

@amrut-prabhu has an article on SQL https://github.com/se-edu/learningresources/pulls/amrut-prabhu

File: _markbind/navigation/mainNav.md [hunk 1]
```diff
@@ -23,56 +23,52 @@
...
+    * [Scalable Development]({{baseUrl}}/contents/scalabilityAndPerformance/scalable-development.html)
+  * Security
+    * [Cross Site Request Forgery (CSRF)]({{baseUrl}}/contents/security/crossSiteRequestForgery/crossSiteRequestForgery.html)
+    * [Cross Site Scripting (XSS)]({{baseUrl}}/contents/security/crossSiteScripting/crossSiteScripting.html)
+    * [Cryptography]({{baseUrl}}/contents/security/cryptography.html)
+    * [HTTPS]({{baseUrl}}/contents/security/Https.html)
+    * [SQL Injection]({{baseUrl}}/contents/security/sqlInjection.html)
+  * Software Architecture
+    * [REST]({{baseUrl}}/contents/softwareArchitecture/RESTArchitecturalStyle.html)
   * Static Analysis
```
Comment (Xenonym) [inline _markbind/navigation/mainNav.md]: Should `Static Analysis` be under `Code Quality`?

**Comment by luyangkenneth** [comment]
@damithc - Gotcha, thanks for pointing that out. For @amrut-prabhu's article (like all the other WIP articles), it still falls under the same category, so I don't foresee needing another round of tweaks to the category structure.

Is there anything else that's making you hesitant about going ahead with this reorg?

File: _markbind/navigation/mainNav.md (hunk 1 above)
Comment (luyangkenneth) [reply _markbind/navigation/mainNav.md]: Yeah I think it makes sense - I previously considered it too but didn't feel super compelled at the time. Glad to know that one more person has similar thoughts! I'll go ahead with moving this 🙂

**Comment by damithc** [comment]
> I think it'll be less painful to incorporate these changes early. I'm also happy to do another smaller round of tweaks if necessary.

It's a choice between one person fixing it in one PR versus many having to fix small parts of it in many PRs :-)

**Comment by damithc** [comment]
`SE Topics` -> `Software Engineering`?

**Comment by luyangkenneth** [comment]
Okay, I think you have a point there - I'll hold back on this until the end of the sem.

**Comment by luyangkenneth** [comment]
Update: I'll continue working on this after the last book chapter PR has been merged.

**Comment by luyangkenneth** [comment]
@damithc This is ready for review :)

**Pull request closed without merging**